hedera-node/src/main/java/com/hedera/services/state/expiry/UniqueTokensListMutation.java

```diff
@@ -17,75 +17,73 @@
 
 import static com.hedera.services.utils.NftNumPair.MISSING_NFT_NUM_PAIR;
 
-import com.hedera.services.state.merkle.MerkleUniqueToken;
-import com.hedera.services.utils.EntityNumPair;
+import com.hedera.services.state.migration.UniqueTokenAdapter;
+import com.hedera.services.state.migration.UniqueTokenMapAdapter;
+import com.hedera.services.store.models.NftId;
 import com.hedera.services.utils.MapValueListMutation;
-import com.swirlds.merkle.map.MerkleMap;
 import org.jetbrains.annotations.Nullable;
 
-public class UniqueTokensListMutation
-        implements MapValueListMutation<EntityNumPair, MerkleUniqueToken> {
+public class UniqueTokensListMutation implements MapValueListMutation<NftId, UniqueTokenAdapter> {
 
-    final MerkleMap<EntityNumPair, MerkleUniqueToken> uniqueTokens;
+    final UniqueTokenMapAdapter uniqueTokens;
 
-    public UniqueTokensListMutation(
-            final MerkleMap<EntityNumPair, MerkleUniqueToken> uniqueTokens) {
+    public UniqueTokensListMutation(final UniqueTokenMapAdapter uniqueTokens) {
         this.uniqueTokens = uniqueTokens;
     }
 
     @Nullable
     @Override
-    public MerkleUniqueToken get(final EntityNumPair key) {
+    public UniqueTokenAdapter get(final NftId key) {
         return uniqueTokens.get(key);
     }
 
     @Nullable
     @Override
-    public MerkleUniqueToken getForModify(final EntityNumPair key) {
+    public UniqueTokenAdapter getForModify(final NftId key) {
         return uniqueTokens.getForModify(key);
     }
 
     @Override
-    public void put(final EntityNumPair key, final MerkleUniqueToken value) {
+    public void put(final NftId key, final UniqueTokenAdapter value) {
         uniqueTokens.put(key, value);
     }
 
     @Override
-    public void remove(final EntityNumPair key) {
+    public void remove(final NftId key) {
         uniqueTokens.remove(key);
     }
 
     @Override
-    public void markAsHead(final MerkleUniqueToken node) {
+    public void markAsHead(final UniqueTokenAdapter node) {
         node.setPrev(MISSING_NFT_NUM_PAIR);
     }
 
     @Override
-    public void markAsTail(final MerkleUniqueToken node) {
+    public void markAsTail(final UniqueTokenAdapter node) {
         node.setNext(MISSING_NFT_NUM_PAIR);
     }
 
     @Override
-    public void updatePrev(final MerkleUniqueToken node, final EntityNumPair prev) {
-        node.setPrev(prev.asNftNumPair());
+    public void updatePrev(final UniqueTokenAdapter node, final NftId prev) {
+        node.setPrev(prev.asEntityNumPair().asNftNumPair());
```

Comment:
Is it worth having a direct `prev.asNftNumPair()` view?

Reply:
yea.. that would be a good idea. Let me take a look at modifying this.